docs/framing.md

```diff
@@ -2,6 +2,46 @@
 
 Framing wraps messages with headers and checksums so receivers can identify message boundaries and verify integrity.
 
+## Frame Format Definitions
+
+All supported frame formats are defined in [`examples/frame_formats.proto`](../examples/frame_formats.proto). This file provides:
+
+- Protocol Buffer definitions for each frame format
+- Enumeration of all frame format types
+- Configuration message for runtime format selection
+
+## Recommended Frame Formats
+
+| Format | Preamble | Header | CRC | Length | Total Overhead | Use Case |
+|--------|----------|--------|-----|--------|----------------|----------|
+| NoFormat | 0 | 0 | 0 | None | 0 | Trusted links, nested protocols |
+| BasicFrame | 1 | 2 | 2 | None | 4 | Standard reliable communication |
+| BasicFrameLen8 | 1 | 3 | 2 | 8-bit | 5 | Variable length messages |
+
+## All Frame Formats
```

Comment:
Lets have Three Frame format sections in total. 

Recommended, Additional recommended (Needs better name) and
Parametric Frame Formats.